presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/db/DbManagerSpecProvider.java

```diff
@@ -0,0 +1,156 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.resourceGroups.db;
+
+import com.facebook.presto.resourceGroups.ManagerSpec;
+import com.facebook.presto.resourceGroups.ResourceGroupIdTemplate;
+import com.facebook.presto.resourceGroups.ResourceGroupSelector;
+import com.facebook.presto.resourceGroups.ResourceGroupSpec;
+import com.facebook.presto.resourceGroups.SelectorSpec;
+import com.facebook.presto.resourceGroups.reloading.ManagerSpecProvider;
+import com.facebook.presto.resourceGroups.reloading.ReloadingManagerSpec;
+import com.facebook.presto.resourceGroups.reloading.ReloadingResourceGroupConfig;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.inject.Inject;
+import io.airlift.units.Duration;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+public class DbManagerSpecProvider
+        implements ManagerSpecProvider
+{
+    private final ResourceGroupsDao dao;
+    private final String environment;
+
+    @Inject
+    public DbManagerSpecProvider(ResourceGroupsDao dao, @ForEnvironment String environment, ReloadingResourceGroupConfig config)
+    {
+        this.dao = requireNonNull(dao, "dao is null");
+        this.environment = requireNonNull(environment, "environment is null");
+        this.dao.createResourceGroupsGlobalPropertiesTable();
+        this.dao.createResourceGroupsTable();
+        this.dao.createSelectorsTable();
+        if (config.getExactMatchSelectorEnabled()) {
+            this.dao.createExactMatchSelectorsTable();
+        }
+    }
+
+    @Override
+    public synchronized ReloadingManagerSpec getSpec()
+    {
+        // New resource group spec map
+        Map<ResourceGroupIdTemplate, ResourceGroupSpec> resourceGroupSpecs = new HashMap<>();
+        // Set of root group db ids
+        Set<Long> rootGroupIds = new HashSet<>();
+        // Map of id from db to resource group spec
+        Map<Long, ResourceGroupSpec> resourceGroupSpecMap = new HashMap<>();
+        // Map of id from db to resource group template id
+        Map<Long, ResourceGroupIdTemplate> resourceGroupIdTemplateMap = new HashMap<>();
+        // Map of id from db to resource group spec builder
+        Map<Long, ResourceGroupSpecBuilder> recordMap = new HashMap<>();
+        // Map of subgroup id's not yet built
```

Comment:
comments above each line is not needed and don't add a lot of value to the reader.
for e.g.
```
     // Set of root group db ids
     Set<Long> rootGroupIds = new HashSet<>();
```
The comment here just reiterates what the variable name already conveys.

Please remove such un-necessary comments.